### ISSUE TITLE
Ending support for IE 11 after v3.0

### ISSUE THREAD

**Issue by joeyparrish** [google/shaka-player#2339]
**TL;DR: Shaka Player will no longer support IE11 beyond v3.0.x**

Shaka Player has supported IE 11 [since v1.6.0](https://github.com/google/shaka-player/blob/master/CHANGELOG.md#160-2015-11-17), added in [PR #176](https://github.com/google/shaka-player/pull/176) and released a little more than 4 years ago.  At that time, IE 11 usage was significantly higher than Edge, and it was still a very important target for many providers.

Since that time, IE 11 usage has steadily declined.  We use [browser usage data from caniuse.com](https://caniuse.com/usage-table) to help drive our support decisions, and those data indicate that IE usage has now fallen below Edge usage for the first time since we've been watching.

Even Microsoft maintains that IE 11 is only a ["compatibility solution"](https://www.theverge.com/2019/2/8/18216767/microsoft-internet-explorer-warning-compatibility-solution) for outdated web apps.  We don't think Shaka Player fits that description.  With the introduction of a [Chromium-based Edge on January 15, 2020](https://www.theverge.com/2019/11/4/20942038/microsoft-edge-chromium-release-date-new-logo-features) with builds for Windows 7 & 8, there is now even less reason to continue supporting IE 11.

IE 11 support has only gotten more expensive for us.  There are over 91 references to IE workarounds and restrictions in our codebase.  We have three custom polyfills totalling 800+ lines just for IE.  IE has gotten even more problematic for us since the introduction of the Shaka Player UI in v2.5, since one of IE's weakest points is its outdated CSS implementation.

For these reasons, the upcoming v3.0 release of Shaka Player will be the last release branch with IE11 support.  When we stop making maintenance releases for v3.0, we will remove IE from our test lab and our list of supported platforms.

Please feel free to leave your feedback, positive or negative, on this GitHub issue.  Thanks for reading!

-Joey, on behalf of the Shaka Player Team

**Comment by alexandercerutti** [comment]
I think it could be useful to add a link to this issue on the main README.md on note number 4 (IE) of the [compatibility table](https://github.com/google/shaka-player#platform-and-browser-support-matrix). What do you think about it?

**Comment by joeyparrish** [comment]
Good idea!  I'll do that.

**Comment by joeyparrish** [comment]
Please note that although the plan & timeline for IE deprecation has not changed, we have decided to rename v2.6 to v3.0.

**Comment by avelad** [comment]
@joeyparrish In which version is this issue going to be solved, in 3.1 or in 4.0? After seeing the post of version 4.0 and the follow-up of a semantic version, I would like to clarify this.

**Comment by joeyparrish** [comment]
Excellent question.  We had decided to drop support for IE before the switch to v3.0, and with the transition to v3.0, also announced that we would end IE support in v3.1.  Apologies if this is confusing (or cheating) in terms of semantic versioning.

**Comment by ismena** [comment]
Closed by #3309
